utils/miller/wrapper.py

```
@@ -0,0 +1,81 @@
+        break
+
+# Delimiter
+for out_format in io_formats:
+    if get_format(snakemake.output[0]) == out_format:
+        io_opts += f" --o{out_format}"
+        break
+
+
+if is_arg("cat", extra):
+    # For cat operations, use all input files
+    inputs = snakemake.input
```

Ruff (0.8.2)

Undefined name `snakemake`

(F821)